classes/robot/crawler.php

```diff
@@ -263,17 +277,21 @@ public function mark_for_crawl($baseurl, $url) {
      *
      * @return timestamp of crawl start
      */
-    public function get_crawlstart() {
-        return property_exists($this->config, 'crawlstart') ? $this->config->crawlstart : 0;
+    public static function get_crawlstart() {
+        self::init_config();
+        $crawlstart = self::$config->crawlstart;
```

Comment:
If you are calling self::init_config(); in multiple places this is a code smell that something isn't right. If you made the config a getter instead of an attribute you wouldn't have this problem, 

ie instead of:

        self::init_config();
        $crawlstart = self::$config->crawlstart;

you would just have:
 
        $crawlstart = self::get_config()->crawlstart;

This is what we discussed yesterday so why did you do something else?